File: AutoInputOnsite - 2013/AutoInputOnsite/Menu.vb
Imports System.ComponentModel
Imports System.Runtime.InteropServices
Imports System.Text
Imports System.Threading.Thread
Imports System.Configuration

Public Class Menu

    Public fra As AutoImportCsv
    Public fra2 As AutoImportNouhinsyo
    Public Ie As SHDocVw.InternetExplorerMedium
    Private WithEvents BackgroundWorker As BackgroundWorker

    ''' <summary>
    ''' 見出＆明細作成
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub btnMs_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnMs.Click
        StartAllJunbi()
        DoAutoImportCsv()
        kanryou()
    End Sub
    Private Sub DoAutoImportCsv()
        fra = New AutoImportCsv
        fra.Ie = Ie
        If cbHyouji.Checked Then fra.Show()
        fra.DoAll()
        fra.Close()
        pb1.Value = 100
        fra.Dispose()
        fra = Nothing
    End Sub


    ''' <summary>
    ''' 納期指定発注
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub btnNouhin_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnNouhin.Click

        StartAllJunbi()
        AutoImportNouhinsyo()
      kanryou()
    End Sub
    Private Sub AutoImportNouhinsyo()
        fra2 = New AutoImportNouhinsyo
        fra2.Ie = Ie
        If cbHyouji.Checked Then
            fra2.Show()
        End If
        fra2.insatu = Me.cbInsatu.Checked
        fra2.DoAll()
        fra2.Close()
        fra2.Dispose()
        fra2 = Nothing
        pb2.Value = 100
    End Sub

    ''' <summary>
    ''' 見出＆明細作成  AND  納期指定発注
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub btnAll_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAll.Click
        StartAllJunbi()
        DoAutoImportCsv()
        System.Threading.Thread.Sleep(500)
        AutoImportNouhinsyo()
        kanryou()
    End Sub

    ''' <summary>
    ''' 完了
    ''' </summary>
    ''' <remarks></remarks>
    Private Sub kanryou()

        Try
            Ie.Quit()
        Catch ex As Exception

        End Try
        MsgBox("完了")
    End Sub

    ''' <summary>
    ''' Start 準備
    ''' </summary>
    ''' <remarks></remarks>
    Private Sub StartAllJunbi()

re:
        Try
            Ie = Nothing
            Ie = New SHDocVw.InternetExplorerMedium
        Catch ex As Exception
            System.Threading.Thread.Sleep(2000)
            GoTo re
        End Try

        If BackgroundWorker IsNot Nothing Then BackgroundWorker.Dispose()
        BackgroundWorker = New BackgroundWorker
        BackgroundWorker.RunWorkerAsync()

        Timer1.Stop()
        Timer1.Start()
        pb1.Value = 0
        pb2.Value = 0

    End Sub

    ''' <summary>
    ''' Pro bar 
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        If fra IsNot Nothing Then
            pb1.Value = fra.ProBar
        End If
        If fra2 IsNot Nothing Then
            pb2.Value = fra2.ProBar
        End If
    End Sub

    ''' <summary>
    ''' Menu_Disposed
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub Menu_Disposed(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Disposed
        Try
            Ie.Quit()
        Catch ex As Exception
        End Try
    End Sub

    ''' <summary>
    ''' Menu_Load
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub Menu_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            Ie.Quit()
        Catch ex As Exception
        End Try
    End Sub

    ''' <summary>
    ''' Windows Object Something
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    ''' <remarks></remarks>
    Private Sub BackgroundWorker_DoWork(ByVal sender As Object, ByVal e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker.DoWork
        Dim com As New Com("")
        com.NewWindowsCom()
    End Sub

End Class

File: AutoInputOnsite C/AutoInputOnsite/AutoInputOnsite/Menu.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Runtime.InteropServices;
using System.Threading;
using System.Configuration;
using AutoInputOnsite;
using Microsoft.VisualBasic;
namespace AutoInputOnsite
{
    public partial class Menu : Form
    {
        public Menu()
        {
            InitializeComponent();
        }
        public AutoImportCsv FormImportCsv;
        public AutoImportNouhinsyo FormNouki;
        public SHDocVw.InternetExplorerMedium Ie;
        private BackgroundWorker BackgroundWorker;


        // Menu_Load
        private void Menu_Load(System.Object sender, System.EventArgs e)
        {
            QuitIE();
        }

        // Menu_Disposed
        private void Menu_Disposed(object sender, System.EventArgs e)
        {
            QuitIE();
        }


        #region 見出＆明細作成

            //見出＆明細作成 Click
            private void btnMs_Click_1(object sender, EventArgs e)
            {
                this.btnAll.Enabled = false;
                this.btnMs.Enabled = false;
                this.btnNouhin.Enabled = false;
                this.cbInsatu.Enabled = false;
                this.cbWeb.Enabled = false;

                StartAllJunbi();
                DoAutoImportCsv();
                QuitIeAndMsgOk();
            }

            //見出＆明細作成 実行
            private void DoAutoImportCsv()
            {
                FormImportCsv = new AutoImportCsv();
                FormImportCsv.Ie = Ie;
                FormImportCsv.IeVisible = this.cbWeb.Checked;
                if (cbHyouji.Checked)
                {
                    FormImportCsv.Show();
                }
                FormImportCsv.DoAll();
                FormImportCsv.Close();
                pb1.Value = 100;
                FormImportCsv.Dispose();
                FormImportCsv = null;
            }

        #endregion

        #region 納期指定発注
            //納期指定発注 ボタンCLICK
            private void btnNouhin_Click(System.Object sender, System.EventArgs e)
            {
                this.btnAll.Enabled = false;
                this.btnMs.Enabled = false;
                this.btnNouhin.Enabled = false;
                this.cbInsatu.Enabled = false;
                this.cbWeb.Enabled = false;

                StartAllJunbi();
                AutoImportNouhinsyo();
                QuitIeAndMsgOk();
            }
            //納期指定発注 実行
            private void AutoImportNouhinsyo()
            {
                FormNouki = new AutoImportNouhinsyo();
                FormNouki.Ie = Ie;
                try
                {
                    FormNouki.IeVisible = this.cbWeb.Checked;
                    if (cbHyouji.Checked)
                    {
                        FormNouki.Show();
                    }
                }
                catch (Exception)
                {
                }
                FormNouki.insatu = this.cbInsatu.Checked;
                FormNouki.DoAll();
                FormNouki.Close();
                FormNouki.Dispose();
                FormNouki = null;
                pb2.Value = 100;
            }
        #endregion

        #region 見出＆明細作成  AND  納期指定発注 ALL
            /// 見出＆明細作成  AND  納期指定発注 ALL
            private void btnAll_Click_1(System.Object sender, System.EventArgs e)
            {
                this.btnAll.Enabled = false;
                this.btnMs.Enabled = false;
                this.btnNouhin.Enabled = false;
                this.cbInsatu.Enabled = false;
                this.cbWeb.Enabled = false;

                //見出＆明細作成
                StartAllJunbi();
                DoAutoImportCsv();
                //納期指定発注
                StartAllJunbi();
                pb1.Value = 100;
                AutoImportNouhinsyo();
                QuitIeAndMsgOk();
            }
        #endregion

        #region 共通

            /// <summary>
            /// Start 準備 
            ///   1.Close Ie
            ///   2.Init BackgroundWorker
            ///   3.Init Timer 
            /// </summary>
            /// <remarks></remarks>
            private void StartAllJunbi()
            {
                CloseIE();
            re:
                try
                {
                    Ie = null;
                }
                catch (Exception)
                {
                    System.Threading.Thread.Sleep(2000);
                    goto re;
                }
                Ie = new SHDocVw.InternetExplorerMedium();
                if (ReferenceEquals(BackgroundWorker, null))
                {
                    BackgroundWorker = new BackgroundWorker();
                    BackgroundWorker.DoWork += BackgroundWorker_DoWork;
                    BackgroundWorker.RunWorkerAsync();
                }
                else
                {
                    if (!BackgroundWorker.IsBusy)
                    {
                        BackgroundWorker.RunWorkerAsync();
                    }
                    else
                    {
                        BackgroundWorker.Dispose();
                        BackgroundWorker = new BackgroundWorker();
                        BackgroundWorker.DoWork += BackgroundWorker_DoWork;
                        BackgroundWorker.RunWorkerAsync();
                    }
                }

                Timer1.Stop();
                Timer1.Start();
                pb1.Value = 0;
                pb2.Value = 0;
            }

            //IEXPLORE 閉じる
            public void CloseIE()
            {
                try
                {
                    System.Diagnostics.Process[] myProcesses;
                    myProcesses = System.Diagnostics.Process.GetProcessesByName("IEXPLORE");
                    foreach (System.Diagnostics.Process instance in myProcesses)
                    {
                        instance.CloseMainWindow();
                    }
                }
                catch (Exception)
                {
                }
            }

            /// <summary>
            /// 完了
            /// </summary>
            /// <remarks></remarks>
            private void QuitIeAndMsgOk()
            {
                QuitIE();
                this.btnAll.Enabled = true;
                this.btnMs.Enabled = true;
                this.btnNouhin.Enabled = true;
                this.cbInsatu.Enabled = true;
                this.cbWeb.Enabled = true;

                MessageBox.Show("完了");
            }
            
            //Quit Ie
            private void QuitIE()
            {
                try
                {
                    Ie.Quit();
                }
                catch (Exception)
                {
                }
            }

            // Timer1
            private void Timer1_Tick_1(System.Object sender, System.EventArgs e)
            {
                if (FormImportCsv != null)
                {
                    pb1.Value = FormImportCsv.ProBar;
                }
                if (FormNouki != null)
                {
                    pb2.Value = FormNouki.ProBar;
                }
            }

            // Windows Object Something
            private void BackgroundWorker_DoWork(object sender, System.ComponentModel.DoWorkEventArgs e)
            {
                Com com = new Com("");
                com.WindowsInputThread();
            }

        #endregion

    }
}